nemo_rl/environments/genrm_environment.py

```diff
@@ -0,0 +1,277 @@
+            )
+            # Validate format and extract scores
+            print("assistant_response: ", assistant_response)
+            print("extracted results: ")
+            if individual_scores is not None:
+                print(individual_scores)
+            if preference_ranking is not None:
+                print(preference_ranking)
+            print()
+            
+            
+            if not is_valid_format:
+                # Apply format penalty
+                reward = self.format_penalty
+                rewards.append(float(reward))
+                observations.append({
+                    "role": "environment",
+                    "content": f"Format violation penalty applied. Error: {error_message}. Reward: {reward}"
+                })
+                
+            else:
+                reward = self.format_penalty
+                try:
+                    if meta["num_responses"] == 1:
+                        if meta["helpfulness_1"] is not None and individual_scores:
+                            reward = -self.distance_abs(individual_scores[0], meta["helpfulness_1"])
+                    
+                    elif meta["num_responses"] == 2:
+                        # Calculate distance for both individual scores
+                        if meta["helpfulness_1"] is not None and individual_scores and len(individual_scores) == 2 and meta["preference_ranking"] is not None and preference_ranking:
+                            reward = - self.distance_abs(individual_scores[0], meta["helpfulness_1"])  - self.distance_abs(individual_scores[1], meta["helpfulness_2"]) - self.distance_abs(preference_ranking, meta["preference_ranking"])
+                
+                except Exception as e:
+                    logging.error(f"Error processing response: {e}")
+
+                rewards.append(float(reward))
+                observations.append({
+                    "role": "environment",
+                    "content": f"Format correct. GenRM evaluation complete. Reward: {reward}"
+                })
+        
+        rewards_tensor = torch.tensor(rewards, dtype=torch.float32)
+        terminateds = torch.ones_like(rewards_tensor, dtype=torch.bool)
+        
+        return EnvironmentReturn(
+            observations=observations,
+            metadata=metadata,
+            next_stop_strings=[None] * len(message_log_batch),
+            rewards=rewards_tensor,
+            terminateds=terminateds,
+        )
+    
+    def global_post_process_and_metrics(self, batch: BatchedDataDict) -> tuple[BatchedDataDict, dict]:
+        """Post processing and metrics calculation."""
+        rewards = batch.get("rewards", torch.tensor([]))
+        num_samples = len(batch.get("idx", []))
+        
+        if len(rewards) == 0:
+            return batch, {}
+        
+        # Convert rewards to numpy for easier computation
+        rewards_np = rewards.numpy() if hasattr(rewards, 'numpy') else np.array(rewards)
```

Comment:
What's easier? Wouldn't it be pretty much the same with `rewards` as a PyTorch CPU tensor?